### ISSUE TITLE
Citus Support: Inherit triggers for template table

### ISSUE THREAD

**Issue by jmealo** [pgpartman/pg_partman#358]
Hello,

I've been enjoying pg_partman, it's a great piece of kit! However, I've hit a rather large snag for my use case:

**Postgres 13:**
If you define a trigger on your parent table, you cannot `DROP` it on a child table as the parent depends on it.
This `DROP` of indexes and triggers occurs in Citus when converting a table's access method from hot/row to cold/columnar.

It would be ideal if one could define triggers on the template table as one can with indexes.

It's recommended/common to use pg_partman with Citus. If I recall correctly: Citus and Fluentbit both suggest a pattern where Postgres ingests JSON data, and an `INSERT` trigger populates some columns/tables from that data. I've used it in the past and it has worked well. I think that it'll continue to pick up steam with the community.

**TLDR; Users with triggers on tables cannot convert pg_partman partitioned tables to columnar storage.**

**Comment by keithf4** [comment]
I've tried working on trigger inheritance in the past and it was trickier for me than I anticipated and I knew that native partitioning was working on it, so I'd kind of put it aside for now.

This sounds different, though, because you're basically trying to define a trigger to exist on all the child tables that doesn't ever exist on the parent. Not sure how that would ever be supported natively. While the purpose of the the template table is to handle things not currently handled in native, the long term hope is that eventually it wouldn't be needed anymore. Right now, if using PG12+, the only thing it handles now is unique indexes and if global indexes are ever made, this would be able to be dropped. Adding this trigger inheritance would essentially make the template table a standard part of partitioning, which I was kind of trying to avoid if possible. It could also be made even trickier since now that trigger inheritance is supported natively, it means there's two places to define triggers to inherit that are doing it in two completely different methods.

So while Citus is recommending pg_partman, and I do appreciate it, this really sounds like an issue unique to Citus when converting to their custom data type, so you may want to speak to them if you haven't yet.

If it does turn out that there are certain in-demand features that never will be supported natively, I can see about revisiting this in the future.

**Comment by keithf4** [comment]
Closing this issue for now. If you have any further questions on this, please feel free to ask.
